test/e2e/simple_preemptor/simple_preemptor_test.go

```diff
@@ -0,0 +1,182 @@
+			sleepRespPod, err := kClient.CreatePod(sleepObj, dev)
+			gomega.Ω(err).NotTo(gomega.HaveOccurred())
+
+			// Wait for pod to move to running state
+			err = kClient.WaitForPodBySelectorRunning(dev,
+				fmt.Sprintf("app=%s", sleepRespPod.ObjectMeta.Labels["app"]),
+				60)
+			gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		}
+		// assert sleeppod2 is killed
+		err := kClient.WaitForPodEvent(dev, "sleepjob2", "Killing", 120)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+	})
+
+	ginkgo.It("Verify_simple_preemption. Use case: When 3 sleep pods (2 opted out, regular) are running, regular pod should be victim to free up resources for 4th sleep pod", func() {
+
+		// Define sleepPod
+		sleepPodConfigs := k8s.SleepPodConfig{Name: "sleepjob", NS: dev, Mem: 2000, Time: 600, RequiredNode: "yk8s-worker"}
+		sleepPod2Configs := k8s.SleepPodConfig{Name: "sleepjob2", NS: dev, Mem: 2000, Time: 600, RequiredNode: "yk8s-worker"}
+		sleepPod3Configs := k8s.SleepPodConfig{Name: "sleepjob3", NS: dev, Mem: 2000, Time: 600, RequiredNode: "yk8s-worker"}
+
+		// random uid in pod spec to ensure this Non-opted pod doesn't become originator
+		sleepPod4Configs := k8s.SleepPodConfig{Name: "sleepjob4", NS: dev, Mem: 2500, Time: 600, Optedout: true, UID: types.UID("ddd")}
+		sleepPod5Configs := k8s.SleepPodConfig{Name: "sleepjob5", NS: dev, Mem: 2500, Time: 600, Optedout: false}
+		sleepPod6Configs := k8s.SleepPodConfig{Name: "sleepjob6", NS: dev, Mem: 2500, Time: 600, Optedout: false}
+		sleepPod7Configs := k8s.SleepPodConfig{Name: "sleepjob7", NS: dev, Mem: 2500, Time: 600, RequiredNode: "yk8s-worker2"}
+
+		for _, config := range []k8s.SleepPodConfig{sleepPodConfigs, sleepPod2Configs,
+			sleepPod3Configs, sleepPod4Configs, sleepPod5Configs, sleepPod6Configs} {
+			ginkgo.By("Deploy the sleep pod " + config.Name + " to the development namespace")
+			sleepObj, podErr := k8s.InitSleepPod(config)
+
+			Ω(podErr).NotTo(gomega.HaveOccurred())
+			sleepRespPod, err := kClient.CreatePod(sleepObj, dev)
+			gomega.Ω(err).NotTo(gomega.HaveOccurred())
+			// Wait for pod to move to running state
+			err = kClient.WaitForPodBySelectorRunning(dev,
+				fmt.Sprintf("app=%s", sleepRespPod.ObjectMeta.Labels["app"]),
+				240)
+			gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		}
+
+		ginkgo.By("Deploy the sleep pod " + sleepPod7Configs.Name + " to the development namespace")
+		sleepObj, podErr := k8s.InitSleepPod(sleepPod7Configs)
+
+		Ω(podErr).NotTo(gomega.HaveOccurred())
+		sleepRespPod, err := kClient.CreatePod(sleepObj, dev)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+
+		// Wait for pod to move to running state
+		err = kClient.WaitForPodBySelectorRunning(dev,
+			fmt.Sprintf("app=%s", sleepRespPod.ObjectMeta.Labels["app"]),
+			600)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+
+		// assert sleeppod6 is killed
+		err = kClient.WaitForPodEvent(dev, "sleepjob4", "Killing", 1200)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+
+	})
+
+	ginkgo.AfterEach(func() {
```

Comment:
I don't think this is necessary.

`ginkgo.AfterSuite()` calls `kClient.TearDownNamespace()` so all sleep pods will be killed anyway.

This is only needed if you plan to add multiple test cases. Even then, just call `kClient.TearDownNamespace()` which does a proper cleanup.